pycloudlib/lxd/instance.py

```diff
@@ -379,6 +379,32 @@ def wait_for_stop(self):
             time.sleep(1)
         raise TimeoutError
 
+    def _wait_for_instance_start(self):
+        """Wait for the cloud instance to be up.
+
+        LXD VMs need to install systemd units upon initialization. There is
+        no easy way to do this and also enable them on boot, so an LXD VM
+        will reboot as part of its initialization process. It is possible
+        we have connected the VM before this reboot occurs, so then any
+        following SSH connections will fail.
+
+        The VM doesn't accurately report the number of processes until
+        the initialization is fully complete, so block until our number
+        of processes isn't -1.
+        """
+        processes = -1
+        for _ in range(300):
+            try:
+                processes = int(subp(
+                    'lxc list -c N {} -f csv'.format(self.name).split()
```

Comment:
We need to be more strict in any calls to lxc list because it does a prefix match on names which may match multiple VMs in CI with common prefix.

Also in what cases do we need "split" with self.name?

```suggestion
                    ['lxc', 'list', '-cN',  '^{}$'.format(self.name),  '-f', 'csv']
```

Reply:
> Also in what cases do we need "split" with self.name?

The split functions on the whole string (post formatting). Only reason is I find it easier to define the command as a string and then split it afterwords, but defining the command as a list works too.

Reply:
ahh right +1